src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -382,11 +395,15 @@ public function fixTemplates(ContainerBuilder $container, Definition $definition
 
         $definedTemplates = $overwrittenTemplates['view'] + $definedTemplates;
 
-        if ($container->getParameter('sonata.admin.configuration.templates') !== $definedTemplates) {
-            $definition->addMethodCall('setTemplates', [$definedTemplates]);
-        } else {
-            $definition->addMethodCall('setTemplates', ['%sonata.admin.configuration.templates%']);
-        }
+        $templateRegistryId = $serviceId.'.template_registry';
+        $templateRegistryDefinition = $container
+            ->register($templateRegistryId, TemplateRegistry::class)
+            ->addTag('sonata.admin.template_registry')
+            ->setPublic(true); // Temporary fix until we can support service locators
+
+        $templateRegistryDefinition->addArgument($definedTemplates);
```

Comment:
Forget what I said before, I think the other issue I mention might be addressed in another PR. Using the reference was on purpose. Sorry for the back and forth

Reply:
No problem. Shall I revert back?

Reply:
Yes please

Reply:
And we're back. Long live `Undo`.